--- cryptsetup-2.4.3/tests/Makefile.am.orig	2022-01-13 10:23:53.000000000 +0100
+++ cryptsetup-2.4.3/tests/Makefile.am	2022-01-20 10:56:51.933912924 +0100
@@ -27,7 +27,7 @@
 endif
 
 if REENCRYPT
-TESTS += reencryption-compat-test reencryption-compat-test2 luks2-reencryption-test luks2-reencryption-mangle-test
+TESTS += reencryption-compat-test reencryption-compat-test2 luks2-reencryption-test
 endif
 
 if INTEGRITYSETUP
